AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_Atom_MeshOptimizerTest.py

```diff
@@ -0,0 +1,138 @@
+            A predicate function what will be used in wait_for_condition.
+            """
+            return self.isModelReady
+
+        def on_model_ready(self, parameters):
+            self.isModelReady = True
+
+        def wait_for_on_model_ready(self, entityId, mesh_component, model_id):
+            self.isModelReady = False
+            # Connect to the MeshNotificationBus
+            # Listen for notifications when entities are created/deleted
+            self.onModelReadyHandler = azlmbr.bus.NotificationHandler('MeshComponentNotificationBus')
+            self.onModelReadyHandler.connect(entityId)
+            self.onModelReadyHandler.add_callback('OnModelReady', self.on_model_ready)
+            
+            waitCondition = partial(self.model_is_ready_predicate)
+            
+            mesh_component.set_component_property_value(AtomComponentProperties.mesh('Model Asset'), model_id)
+            if TestHelper.wait_for_condition(waitCondition, 20.0):
+                return True
+            else:
+                return False
+
+    with Tracer() as error_tracer:
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
+        TestHelper.init_idle()
+        TestHelper.open_level("Graphics", "base_empty")
+
+        # Test steps begin.
+        # 1. Create a Mesh entity with no components.
+        mesh_entity = EditorEntity.create_editor_entity(AtomComponentProperties.mesh())
+        Report.critical_result(Tests.mesh_entity_creation, mesh_entity.exists())
+
+        # 2. Add a Mesh component to Mesh entity.
+        mesh_component = mesh_entity.add_component(AtomComponentProperties.mesh())
+        Report.critical_result(
+            Tests.mesh_component_added,
+            mesh_entity.has_component(AtomComponentProperties.mesh()))
+
+        # 5. Set Mesh component asset property
+        model_path = os.path.join('testdata', 'objects', 'modelhotreload', 'sphere_5lods.azmodel')
+        model = Asset.find_asset_by_path(model_path)
+        onModelReadyHelper = OnModelReadyHelper()
+        onModelReadyHelper.wait_for_on_model_ready(mesh_entity.id, mesh_component, model.id)
+        Report.result(Tests.model_asset_specified,
+                      mesh_component.get_component_property_value(
+                          AtomComponentProperties.mesh('Model Asset')) == model.id)
+        #mesh_component.set_component_property_value(AtomComponentProperties.mesh('Model Asset'), model.id)
+
+        #Report.result(Tests.model_asset_is_optimized,
+        #              mesh_component.get_component_property_value(
+        #                  AtomComponentProperties.mesh('Use ray tracing')) is False)
+        
+        #pteObj = editor.EditorComponentAPIBus(bus.Broadcast, 'BuildComponentPropertyTreeEditor', component)
+        #BaseClass.check_result(pteObj.IsSuccess(), "BuildComponentPropertyTreeEditor")
+        #pte = pteObj.GetValue()
```

Comment:
commented out code still needed?

Reply:
Moved the hydra test to be part of the existing mesh component test, and got rid of the commented out code.